src/rtc.rs

```diff
@@ -3,131 +3,171 @@
         {
             init_mode(&self.rtc, true);
             {
-                
-                let (ht, hu) = byte_to_bcd2(time.hours as u8);
-                let (mnt, mnu) = byte_to_bcd2(time.minutes as u8);
-                let (st, su) = byte_to_bcd2(time.seconds as u8);
-                self.rtc.tr.write(|w| unsafe {
-                    w.ht().bits(ht)
-                        .hu().bits(hu)
-                        .mnt().bits(mnt)
-                        .mnu().bits(mnu)
-                        .st().bits(st)
-                        .su().bits(su)
-                        .pm()
-                        .clear_bit()
+                match time {
+                    Some(time) => {
+                        let (ht, hu) = byte_to_bcd2(time.hours as u8);
+                        let (mnt, mnu) = byte_to_bcd2(time.minutes as u8);
+                        let (st, su) = byte_to_bcd2(time.seconds as u8);
 
-                });
+                        self.rtc.tr.write(|w| unsafe {
+                            w.ht().bits(ht)
+                                .hu().bits(hu)
+                                .mnt().bits(mnt)
+                                .mnu().bits(mnu)
+                                .st().bits(st)
+                                .su().bits(su)
+                                .pm()
+                                .clear_bit()
+                        });
 
-                self.rtc.cr.modify(|_, w| {
-                    w.fmt()
-                        .bit(time.daylight_savings)
+                        self.rtc.cr.modify(|_, w| {
+                            w.bkp()
+                                .bit(time.daylight_savings)
+                        });
+                    },
+                    None => ()
+                }
+                
+                match date {
+                    Some(date) => {
+                        let (dt, du) = byte_to_bcd2(date.date as u8);
+                        let (mt, mu) = byte_to_bcd2(date.month as u8);
+                        let yr = date.year as u16;
+                        let yr_offset = (yr - 1970_u16) as u8;
+                        let (yt, yu) = byte_to_bcd2(yr_offset);
 
-                });
+                        self.rtc.dr.write(|w| unsafe {
+                            w.dt().bits(dt)
+                                .du().bits(du)
+                                .mt().bit(mt > 0)
+                                .mu().bits(mu)
+                                .yt().bits(yt)
+                                .yu().bits(yu)
+                                .wdu().bits(date.day as u8)
+                        });
+                    },
+                    None => ()
+                }                
             }
             init_mode(&self.rtc, false);
         }
         write_protection(&self.rtc, true);
     }
 
-    pub fn get_time(&self) -> Time {
+    pub fn get_time(&self) -> (Date, Time) {
```

Comment:
Can we rename this to `get_date_time(&self)`?

Reply:
Yes we can.